src/thermobench.cpp

```diff
@@ -203,6 +204,35 @@ void write_arr_csv(FILE *fp, double *arr, int size)
     fprintf(fp, "\n");
 }
 
+void wait_cooldown(char* exec_dir){
+
+    FILE *fp = fopen(state.sensors[0].path, "r");
+    if (fp == NULL)
+        err(1, "Error while opening sensor file: %s\n", state.sensors[0].path);
+
+    char* cmd;
+    asprintf(&cmd,"%s/fan_ctrl.sh 1", exec_dir);
+    if(system(cmd) == -1)
+        err(1, "Error while executing shell command: %s\n", cmd);
+
+    while(1){
+        rewind(fp);
+        int temp = INT_MAX;
+        if(fscanf(fp, "%d", &temp) < 1) continue;
```

Comment:
Does this work? I remember that we had to close and open `/sys/...` files again if we wanted to read the new value. If this work, than we should use the same in `measure_tiemer_cb`.

Reply:
It worked while I was testing it, but I will go with the option currently in measure_tiemer_cb, and only change to this after it has been tested and stable on all our platforms.